solang-server/src/serverutils/backend.rs

```diff
@@ -87,6 +97,693 @@ impl Backend {
+        stmt: &Statement,
+        lookup_tbl: &mut Vec<(u64, u64, String)>,
+        symtab: &sema::symtable::Symtable,
+        ns: &ast::Namespace,
+    ) {
+        match stmt {
+            Statement::VariableDecl(_locs, _, _param, expr) => {
+                if let Some(exp) = expr {
+                    Backend::construct_expr(exp, lookup_tbl, symtab, ns);
+                }
+            }
+            Statement::If(_locs, _, expr, stat1, stat2) => {
+                //let _if_msg = String::from("If(...)");
+                Backend::construct_expr(expr, lookup_tbl, symtab, ns);
+                for st1 in stat1 {
+                    Backend::construct_stmt(st1, lookup_tbl, symtab, ns);
+                }
+                for st2 in stat2 {
+                    Backend::construct_stmt(st2, lookup_tbl, symtab, ns);
+                }
+            }
+            Statement::While(_locs, _blval, expr, stat1) => {
+                Backend::construct_expr(expr, lookup_tbl, symtab, ns);
+                for st1 in stat1 {
+                    Backend::construct_stmt(st1, lookup_tbl, symtab, ns);
+                }
+            }
+            Statement::For {
+                loc: _,
+                reachable: _,
+                init,
+                cond,
+                next,
+                body,
+            } => {
+                if let Some(exp) = cond {
+                    Backend::construct_expr(exp, lookup_tbl, symtab, ns);
+                }
+                for stat in init {
+                    Backend::construct_stmt(stat, lookup_tbl, symtab, ns);
+                }
+                for stat in next {
+                    Backend::construct_stmt(stat, lookup_tbl, symtab, ns);
+                }
+                for stat in body {
+                    Backend::construct_stmt(stat, lookup_tbl, symtab, ns);
+                }
+            }
+            Statement::DoWhile(_locs, _blval, stat1, expr) => {
+                Backend::construct_expr(expr, lookup_tbl, symtab, ns);
+                for st1 in stat1 {
+                    Backend::construct_stmt(st1, lookup_tbl, symtab, ns);
+                }
+            }
+            Statement::Expression(_locs, _, expr) => {
+                Backend::construct_expr(expr, lookup_tbl, symtab, ns);
+            }
+            Statement::Delete(_locs, _typ, expr) => {
+                Backend::construct_expr(expr, lookup_tbl, symtab, ns);
+            }
+            Statement::Destructure(_locs, _vecdestrfield, expr) => {
+                Backend::construct_expr(expr, lookup_tbl, symtab, ns);
+            }
+            Statement::Continue(_locs) => {}
+            Statement::Break(_) => {}
+            Statement::Return(_locs, expr) => {
+                for expp in expr {
+                    Backend::construct_expr(expp, lookup_tbl, symtab, ns);
+                }
+            }
+            Statement::Emit {
+                loc: _,
+                event_no: _,
```

Comment:
In the code:
``emit Foo(true, 1);`` Foo should have a hoover-over showing the fields of event Foo.